### PR TITLE
fix: dispatch validation

### LINKED ISSUE #12: Add validation to prevent messages to be published in wrong topics
Our [Dispatch method](https://github.com/matheusmosca/walrus/blob/main/domain/entities/topic.go#L37) is dispatching any message incoming from the function's parameter. So we need to validate if this message really belongs to this topic. You can create a new error for this validation in [errors.go](https://github.com/matheusmosca/walrus/blob/main/domain/entities/errors.go)

It is also important to add unit tests for this scenario.

### PR DESCRIPTION
**Pull request by sptGabriel**
this PR add validation to topic dispatch when the message does belong to the topic

closes #12

### CODE REVIEW

File: domain/entities/errors.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package entities
 import "errors"
 
 var (
+	ErrInvalidMessage     = errors.New("this message does not belong to the topic")
```
Comment (matheusmosca) [inline domain/entities/errors.go]: Could you give a better name to this variable? Like `ErrTopicNameDoesNotMatch`. WDYT @sptGabriel?

Comment (sptGabriel) [reply domain/entities/errors.go]: Yes i go fix linter and name, and other strategy to assert error.

File: domain/entities/topic_test.go [hunk 2]
```diff
@@ -101,13 +126,24 @@ func TestDispatch(t *testing.T) {
...
 			wg.Wait()
+
 			for _, msg := range tt.messages {
-				topic.Dispatch(msg)
+				err := topic.Dispatch(msg)
+				if tt.wantError == nil {
+					require.NoError(t, err)
+				}
+
+				assertInvalidMessage(t, err, tt.wantError)
```
Comment (matheusmosca) [inline domain/entities/topic_test.go]:
```suggestion
				assert.ErrorIs(t, err, tt.wantError)
```

Comment (sptGabriel) [reply domain/entities/topic_test.go]: @matheusmosca hey do you like this?

```
func assertDispatchMessages(t *testing.T, topic Topic, messages []vos.Message, wantErr error) {
	for _, msg := range messages {
		err := topic.Dispatch(msg)
		if wantErr == nil {
			require.NoError(t, err)
		}

		assert.ErrorIs(t, err, err)
	}
}
```

the linter warning about: cognitive complexity 11 of func `TestDispatch` is high (> 10) (gocognit) in this approach

File: domain/entities/topic_test.go [hunk 3]
```diff
@@ -101,13 +126,23 @@ func TestDispatch(t *testing.T) {
...
 		})
 	}
 }
 
+func assertDispatchMessages(t *testing.T, topic Topic, messages []vos.Message, wantErr error) {
+	for _, msg := range messages {
+		err := topic.Dispatch(msg)
+		if wantErr == nil {
+			require.NoError(t, err)
+		}
```
Comment (matheusmosca) [inline domain/entities/topic_test.go]:
```suggestion
```

File: domain/entities/topic_test.go [hunk 4]
```diff
@@ -101,13 +126,23 @@ func TestDispatch(t *testing.T) {
...
+
+		assert.ErrorIs(t, err, err)
```
Comment (matheusmosca) [inline domain/entities/topic_test.go]:
```suggestion
		assert.ErrorIs(t, err, tt.wantErr)
```

File: domain/entities/topic_test.go (hunk 2 above)
Comment (matheusmosca) [reply domain/entities/topic_test.go]: Sure, but take a look at my new comments @sptGabriel

File: domain/entities/topic_test.go (hunk 4 above)
Comment (sptGabriel) [reply domain/entities/topic_test.go]: done

**Review by matheusmosca — APPROVED** [review]

**Pull request merged**